lib/src/sql/value/value.rs

```diff
@@ -2134,9 +2134,16 @@ impl Value {
 
 	/// Try to convert this value to a Record or `Thing`
 	pub(crate) fn convert_to_record(self) -> Result<Thing, Error> {
-		match self {
+		match self.clone() {
```

Comment:
Clone is not necessary here. Even if it were, we should always try to delay it as much as possible so that branches that don't need it aren't affected.

You can remove it here and write your new branch as

```rust
Value::Strand(v) => Thing::try_from(v.as_str()).map_err(move |_| Error::ConvertTo {
    from: Value::Strand(v),
    into: "record".into(),
}),
```

Reply:
Thanks!